Add EMIT_EMSCRIPTEN_LICENSE option to emit license info in JS

kripken opened the pull request:
As the person that created this project, I've never thought it's necessary
to add license info in the output JS, since it's compiler output. However,
IANAL. This adds a compiler flag so people that do want to emit the
license info can do so.

Also move all `EMIT_` flags into a single area in the settings file.

For simplicity the emitted license only mentions MIT. I gave Emscripten
two license options back in the day, MIT and LLVM, the latter just to
avoid issues with sharing code with LLVM. That ended up not necessary.
The LLVM license is technically still there, but not worth mentioning in a
`SPDX-License-Identifier` because I hear automatic tools may fail
to parse a dual license (if anyone does want to use the defunct LLVM
license instead of MIT, please let me know).

sbc100 commented on tools/shared.py:
```diff
@@ -2957,6 +2973,13 @@ class JS(object):
 
   global_initializers_pattern = r'/\* global initializers \*/ __ATINIT__.push\((.+)\);'
 
+  emscripten_license = '''/**
```
For multiline comments like this I find them easier to read if you start them with `''\`

kripken replied:
Oh, I didn't know about that, nice!

kripken commented:
I found a bug here - in one case we ended up adding the license twice, due to how wasm2js optimizes only a subset of the JS. Fixed and added testing for that.

I also decided to not try to support this in fastcomp for that reason. The user asking for this isn't using fastcomp anyhow, and the JS pipeline is much more complex there, and fastcomp is deprecated, so doesn't seem worth the work.

sbc100 commented on tests/test_other.py:
```diff
@@ -9698,6 +9698,24 @@ def test_add_emscripten_metadata(self):
+    'O2_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE']), # noqa
+    'O2_js_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '-s', 'WASM=0']), # noqa
+    'O2_closure_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '--closure', '1']), # noqa
+    'O2_closure_js_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '--closure', '1', '-s', 'WASM=0']), # noqa
```
Give that you are testing at different O levels would this make sense in core?

kripken replied:
Yeah, maybe this could go in core. I kind of like explicitly saying what to test in this case though, since it intentionally includes closure. We don't have a core test mode for closure, so we'd need to add it in manually into core, which removes most of the benefit there I think.

sbc100 commented on tests/test_other.py:
```diff
@@ -9698,6 +9698,24 @@ def test_add_emscripten_metadata(self):
+    run_process([PYTHON, EMCC, path_from_root('tests', 'hello_world.c')] + args)
+    with open('a.out.js') as f:
+      js = f.read()
+    self.assertContainedIf('Copyright 2020 Emscripten authors', js, expect_license)
```
I didn't know we had such a function!
